spec/unit/matrix-client.spec.ts

```diff
@@ -2266,13 +2266,11 @@ describe("MatrixClient", function () {
         function roomCreateEvent(newRoomId: string, predecessorRoomId: string): MatrixEvent {
             return new MatrixEvent({
                 content: {
-                    "creator": "@daryl:alexandria.example.com",
                     "m.federate": true,
                     "predecessor": {
                         event_id: "id_of_last_event",
                         room_id: predecessorRoomId,
                     },
-                    "room_version": "9",
```

Comment:
Why is this going away?

Reply:
Yeah, I wasn't sure what the implication of that would be and all the tests still run without it. So I consciously put it in here to cover it during review. Why do you think the room version should stick around here?

Reply:
> Why do you think the room version should stick around here?

Because the PR is called `Use sender instead of content.creator field on m.room.create events` and that seems to have nothing to do with removing room versions from the content (i.e. reverting all rooms under test to v1)

Reply:
> Because the PR is called `Use sender instead of content.creator field on m.room.create events` and that seems to have nothing to do with removing room versions from the content

Yes, I had tried to explain why I included that in https://github.com/matrix-org/matrix-js-sdk/pull/3675#discussion_r1304094694 which I acknowledge happened simultaneously with your comments.

> i.e. reverting all rooms under test to v1

Oh, I had actually assumed we'd fall back to a default, not v1. In re-reading the spec, I'm realizing that the default is defined on a per-HS basis though.

So then this sounds like in order to be able to remove the `content.creator` fields from the fixtures, we'd have to explicitly include `"room_version": 11` instead.

How do you suggest we handle the fixtures in light of the spec changes in room version 11?

Reply:
> Oh, I had actually assumed we'd fall back to a default, not v1. In re-reading the spec, I'm realizing that the default is defined on a per-HS basis though.

No, that's for the `/createRoom` API. A room without a version baked into its (immutable) `m.room.create` is `v1`

https://spec.matrix.org/v1.8/client-server-api/#mroomcreate

> The version of the room. Defaults to "1" if the key does not exist.

> So then this sounds like in order to be able to remove the content.creator fields from the fixtures, we'd have to explicitly include "room_version": 11 instead.

No as the behaviour being version-independent sounds fine to me, I just don't think your PR needs to touch the versions at all.

Reply:
> No, that's for the `/createRoom` API. A room without a version baked into its (immutable) `m.room.create` is `v1`

Ohh, right. 🤯

Reply:
> No as the behaviour being version-independent sounds fine to me, I just don't think your PR needs to touch the versions at all.

Ok, have made it so.